File: Deliverables/php/model/Profilo.php
<?php
	//**********************************************************************************************
    // Creazione dell' oggetto Utente che implementa JsonSerializable per essere inviato tramite json
    // Variabili: nickname, password, email, img_profilo, statistica
    // jsonSerialize() --> utilizzato per la conversione a json
    //**********************************************************************************************
	
    class Profilo implements JsonSerializable {
    	private $nickname;
        private $password;
        private $email;
        private $img_profilo;
        //private $statistica;
        
        public function __construct($nickname, $password, $email, $img_profilo){
        	$this->nickname = $nickname;
            $this->password = $password;
            $this->email = $email;
            $this->img_profilo = $profilo;
            //$this->statistica = $statistica;
        }
        
        public function getNickname(){
        	return $this->nickname;
        }
        
        public function getPassword(){
        	return $this->password;
        }
        
        public function getEmail(){
        	return $this->email;
        }
        
        public function getImgProfilo(){
        	return $this->img_profilo;
        }
        
        public function getStatistica(){
        	return $this->statistica;
        }
        
        public function setNickname($nickname){
        	$this->nickname = $nickname;
        }
        
        public function setPassword($password){
        	$this->password = $password;
        }
        
        public function setEmail($email){
        	$this->email = $email;
        }
        
        public function setImgProfilo($img_profilo){
        	$this->img_profilo = $img_profilo;
        }
        
        public function setStatistica($statistica){
        	$this->statistica = $statistica;
        }
        
        public function jsonSerialize() {
        	return get_object_vars($this);
    	}
    }
?>

File: Deliverables/java/entity/Profilo.java
package entity;

/**
 * Created by GameSoftItalia on 01/12/2016.
 */
public class Profilo {
    private String nickname, password, email;
    private Statisitca statisitca;

    public Profilo(String nickname, String password, String email) {
        this.nickname = nickname;
        this.password = password;
        this.email = email;
    }

    public String getNickname() {
        return nickname;
    }

    public void setNickname(String nickname) {
        this.nickname = nickname;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Statisitca getStatisitca() {
        return statisitca;
    }

    public void setStatisitca(Statisitca statisitca) {
        this.statisitca = statisitca;
    }

    @Override
    public String toString() {
        return "Profilo[Nickname:"+nickname+",Password:"+password+",Email:"+email+",Statistica:"+statisitca+"]";
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) return false;

        if (!(obj instanceof Profilo)) return false;

        Profilo p = (Profilo) obj;
        return ( this.nickname == p.nickname && this.password == p.password && this.email == p.email && this.statisitca.equals(p.statisitca));
    }
}
